apps/aecore/src/aec_vm_chain.erl

```diff
@@ -193,7 +193,7 @@ oracle_extend(Oracle,_Sign, TTL, State) ->
     {ok, Tx} =
         aeo_extend_tx:new(#{oracle_id  => aec_id:create(oracle, Oracle),
                             nonce      => Nonce,
-                            oracle_ttl => {delta, TTL},
+                            oracle_ttl => TTL,
```

Comment:
Extend transaction TTL can only be absolute ([spec](https://github.com/aeternity/protocol/blame/f622f21f6b41fb5860fd9f37e8df2d5ea7bdb705/oracles/oracle_transactions.md#L46), [deserialization](https://github.com/aeternity/epoch/blob/e3296af5398432e937965c81239cffe0a81ac47d/apps/aeoracle/src/aeo_extend_tx.erl#L158)).

If extend from contract can take absolute TTL (as indicated by this line and by [proposed spec update](https://github.com/aeternity/protocol/pull/178/files#diff-9ddeae469112ac9dd44768c45511c6e9R328)), I would expect [internal representation](https://github.com/aeternity/epoch/blob/e3296af5398432e937965c81239cffe0a81ac47d/apps/aeoracle/src/aeo_extend_tx.erl#L40) needs to reflect that. Cc @hanssv  @gorillainduction

Reply:
I can certainly create and execute extend transactions with absolute TTLs, and it looks to me like the http endpoint allows it too (`ttl_decode` rather than `relative_ttl_decode`):

https://github.com/aeternity/epoch/blob/40bd3dd9cf1293bcf252881b8610b385c85f1c18/apps/aehttp/src/aehttp_dispatch_int.erl#L372

I'm not changing anything about oracles, just adding a Sophia interface to what's currently implemented. If we need to restrict extend transactions to relative TTLs, I think that should be done as a separate ticket.

Reply:
> https://github.com/aeternity/epoch/blob/40bd3dd9cf1293bcf252881b8610b385c85f1c18/apps/aehttp/src/aehttp_dispatch_int.erl#L372

Deeper, when serializing tx, I expect that HTTP endpoint fails if TTL is not relative:

https://github.com/aeternity/epoch/blob/4124f84531e40b33857e40c92adcc04865c41630/apps/aeoracle/src/aeo_extend_tx.erl#L144

And when tx is deserialized, relative delta is enforced too:

https://github.com/aeternity/epoch/blob/4124f84531e40b33857e40c92adcc04865c41630/apps/aeoracle/src/aeo_extend_tx.erl#L159